esp-hal/src/spi/master.rs

```diff
@@ -818,10 +829,24 @@ where
             return Err(Error::FifoSizeExeeded);
         }
 
-        self.spi
-            .init_spi_data_mode(cmd.mode(), address.mode(), data_mode);
-        self.spi
-            .write_bytes_half_duplex(cmd, address, dummy, buffer)
+        self.spi.setup_half_duplex(
+            true,
+            cmd,
+            address,
+            false,
+            dummy,
+            buffer.is_empty(),
+            data_mode,
+        );
+
+        if !buffer.is_empty() {
```

Comment:
Do we need this whole dance if buf is empty? Can't we just return before setup?

Reply:
`write_bytes_half_duplex` was already doing this but I'm guessing the reason it still did the whole dance was because it's valid to start an SPI transaction that doesn't use the data phase and only uses the cmd phase and/or address phase,

Reply:
Good point, I should read the code more before asking. Sorry!